From d7fbc55b6c02397b5ad1ab905b418006338acebf Mon Sep 17 00:00:00 2001
From: mse1969 <mse1969@posteo.de>
Date: Sat, 7 Mar 2020 14:55:40 +0100
Subject: [PATCH] 'Pseudonymize' build metadata

Don't write the real Linux build host name and user, but use the env. vars
KBUILD_BUILD_USER / KBUILD_BUILD_HOST, which are also used to describe the
kernel build metadata, if set: A little less info usable for fingerprinting.

Change-Id: Ia1d9c53ca87c9f208b57f9998ba92aa1d2052cd3
---
 core/Makefile            | 2 +-
 core/version_defaults.mk | 2 +-
 tools/buildinfo.sh       | 4 ++--
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/core/Makefile b/core/Makefile
index 404bb54..6e2f4fa 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -236,7 +236,7 @@ $(intermediate_system_build_prop): PRIVATE_BUILD_DESC := $(build_desc)
 # The string used to uniquely identify the combined build and product; used by the OTA server.
 ifeq (,$(strip $(BUILD_FINGERPRINT)))
   ifeq ($(strip $(HAS_BUILD_NUMBER)),false)
-    BF_BUILD_NUMBER := $(USER)$$($(DATE_FROM_FILE) +%m%d%H%M)
+    BF_BUILD_NUMBER := $(KBUILD_BUILD_USER)$$($(DATE_FROM_FILE) +%m%d%H%M)
   else
     BF_BUILD_NUMBER := $(file <$(BUILD_NUMBER_FILE))
   endif
diff --git a/core/version_defaults.mk b/core/version_defaults.mk
index 8c3cb60..6ed6acb 100644
--- a/core/version_defaults.mk
+++ b/core/version_defaults.mk
@@ -289,7 +289,7 @@ ifndef BUILD_NUMBER
   # If no BUILD_NUMBER is set, create a useful "I am an engineering build
   # from this date/time" value.  Make it start with a non-digit so that
   # anyone trying to parse it as an integer will probably get "0".
-  BUILD_NUMBER := eng.$(shell echo $${USER:0:6}).$(shell $(DATE) +%Y%m%d.%H%M%S)
+  BUILD_NUMBER := eng.$(KBUILD_BUILD_USER).$(shell $(DATE) +%Y%m%d.%H%M%S)
   HAS_BUILD_NUMBER := false
 endif
 
diff --git a/tools/buildinfo.sh b/tools/buildinfo.sh
index 9a03fad..0886036 100755
--- a/tools/buildinfo.sh
+++ b/tools/buildinfo.sh
@@ -17,8 +17,8 @@ echo "ro.build.version.min_supported_target_sdk=$PLATFORM_MIN_SUPPORTED_TARGET_S
 echo "ro.build.date=`$DATE`"
 echo "ro.build.date.utc=`$DATE +%s`"
 echo "ro.build.type=$TARGET_BUILD_TYPE"
-echo "ro.build.user=$USER"
-echo "ro.build.host=`hostname`"
+echo "ro.build.user=$KBUILD_BUILD_USER"
+echo "ro.build.host=$KBUILD_BUILD_HOST"
 echo "ro.build.tags=$BUILD_VERSION_TAGS"
 echo "ro.build.flavor=$TARGET_BUILD_FLAVOR"
 if [ -n "$BOARD_BUILD_SYSTEM_ROOT_IMAGE" ] ; then
-- 
2.7.4

